client/components/button/test/index.js

```diff
@@ -0,0 +1,1064 @@
+/**
+ * Created by nitrajka on 09/10/16.
+ */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { expect } from 'chai';
+import { shallow } from 'enzyme';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../';
+import Gridicon from 'components/gridicon';
+
+describe( 'Button', function() {
+	describe( 'not compact', function() {
+		describe( 'button tag', function() {
+			const text = 'Text';
+			const icon = ( <Gridicon size={ 18 } icon="arrow-left" /> );
+			const disabled = 'Disabled';
+
+			describe( 'basic styled', function() {
+				it( 'renders with text', function() {
+					const button = shallow( <Button>{ text }</Button> );
+					expect( button ).to.contain( 'Text' );
+					expect( button.is( '.button' ) ).to.equal( true );
+				} );
+
+				it( 'renders with text and icon', function() {
```

Comment:
I don't know that there's any benefit in this test beyond what's already achieved in the basic render test. Since the `<Button />` component simply renders the children passed to it, we should be reasonably confident that all of the props of descendent components are unaffected.